src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/Completion/CompletionHandler.cs

```diff
@@ -0,0 +1,191 @@
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Utilities;
+using System.Linq;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.EditorConfigSettings.Data;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.EditorConfig.Features
+{
+    [ExportStatelessLspService(typeof(CompletionHandler), ProtocolConstants.EditorConfigLanguageContract), Shared]
+    [Method(Methods.TextDocumentCompletionName)]
+    internal sealed class CompletionHandler : IRequestHandler<CompletionParams, CompletionList?>
+    {
+        private static readonly ImmutableArray<string> _settingNameCommitCharacters = ImmutableArray.Create(new string[] { " ", "=" });
+        private static readonly ImmutableArray<string> _multipleValuesCommitCharacters = ImmutableArray.Create(new string[] { "," });
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CompletionHandler()
+        {
+        }
+
+        public bool MutatesSolutionState => false;
+        public bool RequiresLSPSolution => true;
+
+        public TextDocumentIdentifier GetTextDocumentIdentifier(CompletionParams request) => request.TextDocument;
+
+        public async Task<CompletionList?> HandleRequestAsync(CompletionParams request, RequestContext context, CancellationToken cancellationToken)
+        {
+            var document = context.AdditionalDocument;
+            Contract.ThrowIfNull(document);
+
+            if (request.Context == null)
+            {
+                return null;
+            }
+
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var offset = text.Lines.GetPosition(ProtocolConversions.PositionToLinePosition(request.Position));
+            var textInLine = text.Lines.GetLineFromPosition(offset).ToString();
+            var textToCheck = textInLine[..request.Position.Character].Reverse();
+
+            // Correct syntax is setting_name = setting_value_1, setting_value2, setting_value3... or setting_name = setting_value 
+            // When there exists more then one '=' it is incorrect and we should not suggest any completion 
+            if (textInLine.Count(c => c == '=') > 1)
+            {
+                return null;
+            }
+
+            // Check if we need to display values of the settings
+            // Show completion: |setting_name = (caret)
+            // Show completion: |setting_name = setting_value_1, (caret)
+            // Dont't show completion: | setting_name = setting_value (caret)
```

Comment:
should we show completion on `setting_va(caret)` as well?

Reply:
Yes!